oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -210,6 +196,33 @@ public AccessToken refreshAccessToken() throws IOException {
     return new AccessToken(accessToken, new Date(expiresAtMilliseconds));
   }
 
+  /**
+   * Returns a Google ID Token from the refresh token response.
+   *
+   * @param targetAudience the aud: field the IdToken should include. This is not unused for
+   *     UserCredentials.
+   * @param options list of Credential specific options for for the token. Currently unused for
+   *     UserCredentials.
+   * @throws IOException if the attempt to get an IdToken failed
+   * @return IdToken object which includes the raw id_token, expiration and audience
+   */
+  @Override
+  public IdToken idTokenWithAudience(String targetAudience, List<Option> options)
+      throws IOException {
+    GenericData responseData = doRefreshAccessToken();
+    String idTokenKey = "id_token";
+    if (responseData.containsKey(idTokenKey)) {
+      String idTokenString =
+          OAuth2Utils.validateString(responseData, idTokenKey, PARSE_ERROR_PREFIX);
+      return IdToken.create(idTokenString);
+    }
+
+    throw new IOException(
+        "UserCredentials instance cannot refresh id token unless you are logged"
```

Comment:
That page is not very clear about this, I think we need to write something like

"UserCredentials can obtain an id token only when authentication has been obtained through gcloud running 'gcloud auth login --update-adc' or 'gcloud auth application-default login'. The latter form would not work for Cloud Run, but would still generate an id token.

Reply:
Thanks for suggestion, updated the exception